contracts/core/script/admin/Admin.s.sol

```diff
@@ -234,13 +235,23 @@ contract Admin is Script {
      * @param deployer  The address of the account that will deploy the new implementation.
      */
     function upgradeStaking(address admin, address deployer, bytes calldata data) public {
+        Staking staking = Staking(Predeploys.Staking);
+
+        // read storage pre-upgrade
+        address owner = staking.owner();
+        bool isAllowlistEnabled = staking.isAllowlistEnabled();
+
         vm.startBroadcast(deployer);
         address impl = address(new Staking());
         vm.stopBroadcast();
 
         _upgradeProxy(admin, Predeploys.Staking, impl, data);
 
-        // TODO: add post upgrade tests
+        // assert storage unchanged
+        require(staking.owner() == owner, "owner changed");
+        require(staking.isAllowlistEnabled() == isAllowlistEnabled, "isAllowlistEnabled changed");
+
```

Comment:
Let's test that initializers are disabled on implementation (and fix if they are not)

Reply:
I believe that check is performed globally for all Admin script actions [here](https://github.com/omni-network/omni/blob/8298b12d3104826aa85bba2ac4db87c5d32ec250/contracts/core/script/admin/Admin.s.sol#L388)